File: 0443-string-compression/0443-string-compression.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function(arr) {
    let ans = "";
 let  i = 0, j = 0;

while (j < arr.length) {
    if (arr[i] == arr[j]) {
        j++;
    } else {
        ans += arr[i];
        let len = j - i;   
        if (len > 1) ans += len;

        i = j;
    }
}


ans += arr[i];
let len = j - i;     
if (len > 1) ans += len;

for(i = 0 ; i < ans.length ; i++){
    arr[i] = ans[i]
}

return ans.length;
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] arr) {
       StringBuilder ans = new StringBuilder();
      int i = 0, j = 0;

while (j < arr.length) {
    if (arr[i] == arr[j]) {
        j++;
    } else {
        ans.append( arr[i]);
        int len = j - i;   
        if (len > 1) ans.append(len);

        i = j;
    }
}


ans.append( arr[i]);
int len = j - i;   
if (len > 1) ans.append(len);

for( i = 0 ; i < ans.length(); i++){
    arr[i] = ans.charAt(i);
}

return ans.length();
    }
}